### PR TITLE
disttask: remove scheduler/dispatcher id from logs and refinement

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #46258

Problem Summary:

### What is changed and how it works?

remove scheduler/dispatcher id from logs, the server log itself will identify where we're in

avoid stack overriding in scheduler.OnErr using errors.Trace

some comments

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
run import into task, check logs
```

[2023/09/25 19:48:00.041 +08:00] [INFO] [dispatcher.go:481] [onNextStage] [task-id=90001] [task-type=ImportInto] [current-step=3] [next-step=4]
....
[2023/09/25 19:47:59.141 +08:00] [INFO] [scheduler.go:131] ["scheduler run a step"] [task-id=90001] [step=3] [concurrency=4]

```
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by ywqzzy — APPROVED** [review]

File: disttask/framework/proto/task.go [hunk 1]
```diff
@@ -20,16 +20,58 @@ import (
...
-//  1. succeed/failed:	pending -> running -> succeed/failed
-//  2. canceled:		pending -> running -> canceled
-//  3. rollback:		revert_pending -> reverting -> reverted/revert_failed
-//  4. pause/resume:	pending -> running -> paused -> running
+//		                ┌──────────────────────────────┐
+//		                │           ┌───────┐       ┌──┴───┐
+//		                │ ┌────────►│pausing├──────►│paused│
+//		                │ │         └───────┘       └──────┘
+//		                ▼ │
+//		┌───────┐     ┌───┴───┐     ┌────────┐
```
Comment (Benjamin2037) [inline disttask/framework/proto/task.go:29]: paused -> pending?

**Review by Benjamin2037 — APPROVED** [review]
LGTM

**Pull request merged**